ludwig/hyperopt/execution.py

```diff
@@ -806,44 +787,73 @@ def run_experiment_trial(config, local_hyperopt_dict, checkpoint_dir=None):
+            resources = [self.trial_function_resources, {"CPU": num_cpus, "GPU": num_gpus}]
+
+        resources_per_trial = PlacementGroupFactory(resources)
+        run_experiment_trial_params = tune.with_resources(run_experiment_trial_params, resources_per_trial)
+
         @ray.remote(num_cpus=0)
         def _register(name, trainable):
             register_trainable(name, trainable)
 
         ray.get(_register.remote(f"trainable_func_f{hash_dict(config).decode('ascii')}", run_experiment_trial_params))
 
-        # Note that resume="AUTO" will attempt to resume the experiment if possible, and
-        # otherwise will start a new experiment:
-        # https://docs.ray.io/en/latest/tune/tutorials/tune-stopping.html
-        should_resume = "AUTO" if resume is None else resume
-
         try:
-            analysis = tune.run(
-                f"trainable_func_f{hash_dict(config).decode('ascii')}",
-                name=experiment_name,
-                config={
-                    **self.search_space,
-                    **tune_config,
-                },
-                scheduler=self.scheduler,
-                search_alg=search_alg,
-                num_samples=self.num_samples,
-                keep_checkpoints_num=1,
-                max_failures=1,  # retry a trial failure once
-                resources_per_trial=resources_per_trial,
-                time_budget_s=self.time_budget_s,
-                sync_config=self.sync_config,
-                local_dir=output_directory,
-                metric=metric,
-                mode=mode,
-                trial_name_creator=lambda trial: f"trial_{trial.trial_id}",
-                trial_dirname_creator=lambda trial: f"trial_{trial.trial_id}",
-                callbacks=tune_callbacks,
-                stop=CallbackStopper(callbacks),
-                verbose=hyperopt_log_verbosity,
-                resume=should_resume,
-                log_to_file=True,
-            )
+            tuner_cls = TunerRay210 if not _ray220 else Tuner
+
+            if resume:
+                # Restore from experiment directory
+                tuner = tuner_cls.restore(
+                    path=os.path.join(output_directory, experiment_name), resume_unfinished=True, restart_errored=True
+                )
+            else:
+                tuner = tuner_cls(
+                    f"trainable_func_f{hash_dict(config).decode('ascii')}",
+                    param_space={
+                        **self.search_space,
+                        **tune_config,
+                    },
+                    tune_config=TuneConfig(
+                        metric=metric,
+                        mode=mode,
+                        search_alg=search_alg,
+                        scheduler=self.scheduler,
+                        num_samples=self.num_samples,
+                        time_budget_s=self.time_budget_s,
+                        reuse_actors=True,
```

Comment:
What are the implications of `reuse_actors` here? For GPU training, this can result in GPU memory leaks if we're not careful (https://github.com/ray-project/ray/issues/29624).

One thing we may want to do it set this equal to `not use_gpu`.

Reply:
@arnavgarg1 do you have context here?

Reply:
@tgaddair Good callout, yeah I think we might want to set this to `not use_gpu` to prevent memory leaks. I didn't realize that GPU memory isn't cleaned up on successive runs. Looks like this is fixed in Ray 2.2 so we can probably enable this again once we bump Ludwig to 2.2. Will add a TODO comment